### ISSUE TITLE
Use CMake as a build system

### ISSUE THREAD

**Issue by KOLANICH** [aide/aide#52]

**Comment by oss-assist** [comment]
Any willing to do that? I'd do it and make the pull request.

**Comment by Zugschlus** [comment]
Is there any reason for doing so?

**Comment by KOLANICH** [comment]
1 crossbuild and cross-toolchain compatibility
2 hate to autotools

**Comment by hvhaugwitz** [comment]
At the moment I don't see any reason to move away from autotools. But I'm open for serious arguments.

What are the advantages/disadvantages for me as developer and/or the user of AIDE.

**Comment by oss-assist** [comment]
@hvhaugwitz cmake is sort of a meta build system.
You can generate build config for different build systems like GNU Make and Ninja.
<pre>
Generators
  Unix Makefiles               = Generates standard UNIX makefiles.
  Ninja                        = Generates build.ninja files.
  Watcom WMake                 = Generates Watcom WMake makefiles.
  CodeBlocks - Ninja           = Generates CodeBlocks project files.
  CodeBlocks - Unix Makefiles  = Generates CodeBlocks project files.
  CodeLite - Ninja             = Generates CodeLite project files.
  CodeLite - Unix Makefiles    = Generates CodeLite project files.
  Sublime Text 2 - Ninja       = Generates Sublime Text 2 project files.
  Sublime Text 2 - Unix Makefiles
                               = Generates Sublime Text 2 project files.
  Kate - Ninja                 = Generates Kate project files.
  Kate - Unix Makefiles        = Generates Kate project files.
  Eclipse CDT4 - Ninja         = Generates Eclipse CDT 4.0 project files.
  Eclipse CDT4 - Unix Makefiles= Generates Eclipse CDT 4.0 project files.
</pre>

**Comment by oss-assist** [comment]
@KOLANICH I hate msbuild and its fellows too.

**Comment by KOLANICH** [comment]
@oss-assist , the main reason to use cmake is not a build tool (`make`, `ninja`, `nmake`), but its compiler abstraction layer (I don't have to create a compiler-specific code myself for determining  compiler features for which which abstractions are already implemented in cmake (for example I can ask CMake to require the latest C++ standard and CMake will generate the flags for a compiler itself) ) and out of the box cross-build support (one just creates a toolchain file with paths to needed tools).

**Comment by oss-assist** [comment]
@KOLANICH That's right, CMake is very mature and written in C++ which is a very powerful language.
Better switch to it.
@hvhaugwitz @Zugschlus

**Comment by Zugschlus** [comment]
sounds like marketingblurb to me.
